File: arena-junit/src/main/java/arena/junit/ffi/ArenaBindingError.java
package arena.junit.ffi;
public class ArenaBindingError extends RuntimeException {
  private final ArenaStatus status;
  private final String stateDocument;

  public ArenaBindingError(String message) {
    super(message);
    this.status = null;
    this.stateDocument = null;
  }

  public ArenaBindingError(String message, ArenaStatus status) {
    super(message);
    this.status = status;
    this.stateDocument = null;
  }

  public ArenaBindingError(String message, ArenaStatus status, String stateDocument) {
    super(message);
    this.status = status;
    this.stateDocument = stateDocument;
  }

  public ArenaStatus status() {
    return status;
  }

  public String stateDocument() {
    return stateDocument;
  }
}

File: arena-xunit/src/Ffi/ArenaBindingError.cs
namespace ArenaDotnet.Xunit.Ffi;

public class ArenaBindingError : System.Exception
{
    public string? StateDocument { get; }

    public ArenaBindingError(string message) : base(message) { }
    public ArenaBindingError(string message, System.Exception? inner) : base(message, inner) { }

    public ArenaBindingError(string message, string? stateDocument) : base(message)
    {
        StateDocument = stateDocument;
    }
}
